app/src/main/java/com/kamron/pogoiv/Pokefly.java

```diff
@@ -343,21 +345,29 @@ private static PokemonNameCorrector initCorrectorFromPrefs(PokeInfoCalculator po
 
     @Override
     public int onStartCommand(Intent intent, int flags, int startId) {
-        if (intent != null && intent.hasExtra(KEY_TRAINER_LEVEL)) {
-            trainerLevel = intent.getIntExtra(KEY_TRAINER_LEVEL, 1);
-            statusBarHeight = intent.getIntExtra(KEY_STATUS_BAR_HEIGHT, 0);
-            batterySaver = intent.getBooleanExtra(KEY_BATTERY_SAVER, false);
-            makeNotification(this);
-            createInfoLayout();
-            createIVButton();
-            createArcPointer();
-            createArcAdjuster();
-            /* Assumes MainActivity initialized ScreenGrabber before starting this service. */
-            if (!batterySaver) {
-                screen = ScreenGrabber.getInstance();
-                startPeriodicScreenScan();
-            } else {
-                screenShotHelper = ScreenShotHelper.start(Pokefly.this);
+        if (intent != null) {
```

Comment:
To avoid excessive indentation, I'd prefer writing this as:

```
if (intent == null) {
  return START_STICKY;
}

if (ACTION_STOP.equals(....)) {

} else if ( ) {

}
```

Btw, any reason you don't write it as intent.getAction().equals(ACTION_STOP). Personally, they way you've written it reads a bit odd for me. It's like having to write 5 = i instead of writing i = 5. This last part is not a strong opinion.